src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/DateFieldFormatter.java

```diff
@@ -0,0 +1,167 @@
+        this.dateColumns = getDateColumns(columns);
+        this.fieldAliasMap = fieldAliasMap;
+    }
+
+    /**
+     * Apply the JDBC date format ({@code yyyy-MM-dd HH:mm:ss.SSS}) to date values in the current row.
+     *
+     * @param rowSource The row in which to format the date values.
+     */
+    public void applyJDBCDateFormat(Map<String, Object> rowSource) {
+        for (String columnName : dateColumns) {
+            Object columnOriginalDate = rowSource.get(columnName);
+            if (columnOriginalDate == null) {
+                // Don't try to parse null date values
+                continue;
+            }
+
+            String columnFormat = getFormatForColumn(columnName);
+            if (columnFormat == null) {
+                LOG.warn("Could not determine date format for column {}; returning original value", columnName);
+                continue;
+            }
+            DateFormat format = DateFormat.valueOf(columnFormat.toUpperCase());
+
+            Date date = parseDateString(format, columnOriginalDate.toString());
+            if (date != null) {
+                rowSource.put(columnName, DateFormat.getFormattedDate(date, FORMAT_JDBC));
+            } else {
+                LOG.warn("Could not parse date value; returning original value");
+            }
+        }
+    }
+
+    private String getFormatForColumn(String columnName) {
+        // Handle special cases for column names
+        if (fieldAliasMap.get(columnName) != null) {
+            // Column was aliased, and we need to find the base name for the column
+            columnName = fieldAliasMap.get(columnName);
+        } else if (columnName.split("\\.").length == 2) {
+            // Column is part of a join, and is qualified by the table alias
+            columnName = columnName.split("\\.")[1];
+        }
+        return dateFieldFormatMap.get(columnName);
+    }
+
+    private Set<String> getDateColumns(List<Schema.Column> columns) {
+        return columns.stream()
+            .filter(column -> column.getType().equals(Schema.Type.DATE.nameLowerCase()))
+            .map(Schema.Column::getName)
+            .collect(Collectors.toSet());
+    }
+
+    private Map<String, String> getDateFieldFormatMap(String indexName) {
+        LocalClusterState state = LocalClusterState.state();
+        Map<String, String> formatMap = new HashMap<>();
+
+        String[] indices = indexName.split("\\|");
+        Collection<TypeMappings> typeProperties = state.getFieldMappings(indices)
+            .allMappings();
+
+        for (TypeMappings mappings: typeProperties) {
+            FieldMappings fieldMappings = mappings.firstMapping();
+            for (Map.Entry<String, Map<String, Object>> field : fieldMappings.data().entrySet()) {
+                String fieldName = field.getKey();
+                Map<String, Object> properties = field.getValue();
+
+                if (properties.containsKey("format")) {
```

Comment:
How does multiple date format covered?
https://www.elastic.co/guide/en/elasticsearch/reference/current/date.html#multiple-date-formats

Reply:
Currently not handled; will add handling & testing for this

Reply:
Added handling & tests for multiple formats